### DISCUSSION TITLE
priority signs 'route mode'

### DISCUSSION THREAD

**Discussion by originalfoo** [CitiesSkylinesMods/TMPE#1244] [General]
Just checking all the tools vs. documentation, and on priority signs when holding `Shift` over an icon to apply along route, I'm getting this:

![image](https://user-images.githubusercontent.com/1386719/146840268-b73bbc47-74e6-4531-93cd-7b2630025e0f.png)

Anyone know what's causing that to happen? Bug?

**Comment by originalfoo** [comment]
tl;dr:

* Bend nodes shouldn't break route
* Fork in road should terminate route at that junction
    * EDIT: Is this unexpected behaviour resulting from the `Ctrl+Shift+Click` stuff (high priority road / roundabout policies) which can/should handle forks?

@kianzarrin 

----

Ok, same happpens when doing `Shift` on a segment.

And I just realised, based on the OSD hints panel, that once a junction is selected (ugh, this interaction model for priority signs is whack btw) that `Shift+Click icon` isn't used any more - I'll update wiki docs.

Additionally, I noticed it's stopping at `bend` nodes, which seems confusing to me.

![image](https://user-images.githubusercontent.com/1386719/146843518-30a9e960-9af8-4d69-a6e4-f7f455128c97.png)

So... it looks like it's following road by 'direction' yes?

When it arrives at a fork in the road (ie. "there are two roads going straight ahead from this junction") then IMO it should terminate the route at that junction.

**Reply by kianzarrin** [reply]
I don't remember clearly. probably  a bug. following bend nodes seems to be wise as well. 

also blend nodes should be supported as well (no crossings, enter blocked junction).

**Reply by originalfoo** [reply]
I've updated Priority Signs docs: https://github.com/CitiesSkylinesMods/TMPE/wiki/Priority-Signs

So just to summarise desired changes:

`Shift+Click segment` (cycle priority signs along route):

* Terminate route at forks (or any situation that would result in route splitting in two)
* Don't terminate at bend or blend nodes

`Ctrl+Shift+Click` (apply roundabout or high priority road policies as applicable):

* Forks ... keep existing functionality?
* Don't terminate at bend or blend nodes

**Reply by kianzarrin** [reply]
for the `control+shift+click` there is also this page:
https://github.com/CitiesSkylinesMods/TMPE/wiki/High-Priority-Roads

**Comment by kianzarrin** [comment]
There is a feature freeze going on until we release TMPE. until then I only fix important bugs. 
usually feature freeze should last a few days but this one might last a month or two.
